### ISSUE TITLE
Multi-touch zoom not working on hybrid laptop

### ISSUE THREAD

**Issue by hermanbergwerf** [arose/ngl#53]
I have a Chromebook with both a touch screen and a touchpad. I can zoom using my touchpad but not using multi-touch on the touch screen.

**Comment by arose** [comment]
Is this on http://arose.github.io/ngl/, if not does it work there? Does it work on this site http://threejs.org/examples/#misc_controls_trackball? Are there any messages in the browser console?

**Comment by hermanbergwerf** [comment]
It doesn't work on the three.js site and there are no console errors. The problem is basically that the system default to page zooming when you use multitouch zoom. I just discovered zooming out works (because the page cannot be zoomed out any more). You have to block the event from propagating to other elements/prevent the default action (e.g. `e.preventDefault()`).

**Comment by arose** [comment]
Thanks @hermanbergwerf added some `e.preventDefault()`s, see be33e1455525b2b43dfbba122374b16decc6ed2f. Please let me know if it works for you http://arose.github.io/ngl/.

**Comment by hermanbergwerf** [comment]
Unfortunately it did not work. I shortly looked into it and this oneliner solves the issue.
```js
document.getElementById('viewport').addEventListener('touchmove',function(e){e.preventDefault();});
```
Preventing on scoll is probably not working since multitouch zoom is not considered a scroll event by the browser. You should block the propagation of `touchmove`.

**Comment by arose** [comment]
Cool, I add that later today. Thanks!
 
 
On Tue, Mar 15, 2016, at 11:30 AM, Herman Bergwerf wrote:
> Unfortunately it did not work. I shortly looked into it and this
> oneliner solves the issue.
> document.getElementById('viewport').addEventListener('touchmove',func-
> tion(e){e.preventDefault();});
> Preventing on scoll is probably not working since multitouch zoom is
> not considered a scroll event by the browser. You should block the
> propagation of touchmove.
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub[1]
>
 

Links:

  1. https://github.com/arose/ngl/issues/53#issuecomment-196963459

**Comment by arose** [comment]
Ok, could reproduce this and hopefully fixed it now (73983c5f363c4f7679c0d5b2219d5809c9454f25, 2f139f4d15ea73f2f1df179ae397f15c60012f3d). Thanks again.

**Comment by hermanbergwerf** [comment]
Yep, fixed now (on my Chromebook). Awesome!
